src/components/TaskCard/TaskCard.jsx

```diff
@@ -1,16 +1,27 @@
+import { useDispatch } from "react-redux";
 import PropTypes from "prop-types";
+import { deleteTask } from "src/store/task/actions/taskActions";
+import DeleteButton from "src/components/Buttons/DeleteButton/DeleteButton";
 import "./TaskCard.scss";
 
-function TaskCard({ taskTitle, createdAt }) {
+function TaskCard({ taskId, taskTitle, createdAt }) {
+  const dispatch = useDispatch();
+
+  function handleDeleteTask() {
+    dispatch(deleteTask(taskId));
+  }
+
   return (
     <div className="tasks-container__box">
       <h3>{taskTitle}</h3>
       <p>Created At: {createdAt}</p>
+      <DeleteButton onDelete={handleDeleteTask} />
     </div>
   );
 }
 
 TaskCard.propTypes = {
+  taskId: PropTypes.string.isRequired,
   taskTitle: PropTypes.string.isRequired,
   createdAt: PropTypes.string.isRequired,
```

Comment:
is it a date type? if it is, can we validate if it is an instance of Date class?

Reply:
@ahr-wtag as of now it is stored as a string. That's why it is validated as string.
However, it is converted to date type for the ease of comparing two dates in the next step.